content/blog/case-weights/index.Rmd

```diff
@@ -0,0 +1,310 @@
+---
+output: hugodown::hugo_document
+
+slug: case-weights
+title: Using case weights with tidymodels
+date: 2022-05-01
+author: Max Kuhn
+description: >
+    The tidymodels packages have enabled the use of case weights across many 
+    packages. This post discusses the changes. 
+
+photo:
+  url: https://unsplash.com/photos/iLKK0eFTywU
+  author: Graphic Node
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [deep-dive] 
+tags: [tidymodels parsnip]
+---
+
+<!--
+TODO:
+* [ ] Look over / edit the post's title in the yaml
+* [ ] Edit (or delete) the description; note this appears in the Twitter card
+* [ ] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [ ] Find photo & update yaml metadata
+* [ ] Create `thumbnail-sq.jpg`; height and width should be equal
+* [ ] Create `thumbnail-wd.jpg`; width should be >5x height
+* [ ] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(doMC)
+registerDoMC(cores = 10)
+library(tidymodels)
+tidymodels_prefer()
+theme_set(theme_bw())
+```
+
+After quite a bit of time, the tidymodels packages have enabled the use of case weights. There has been a ton of work and some technical hurdles to overcome. The diversity of the types of weights, and how they should be used, is very complex but I think that we've come up with a solution that is fairly straight-forward for the users. 
+
+## What are case weights?
+
+Case weights are non-negative numbers that are used to impact how much each observation influences a model fit. 
+
+There are different types of case weights whose terminology can be very different across problem domains. Here are some examples: 
+
+* **Frequency weights** are integers that denote how many times a particular row of data has been observed. They help compress redundant rows into a smaller format. 
+* **Importance weights** focus on how much each row of the data set should influence the model fit. These can be based on data or arbitrarily set to achieve some goal. 
+* When survey respondents have different probabilities of selection, (inverse) **probability weights** can help remove bias from the results of a data analysis. 
+* If a data point has an associated precision, **analytic weighting** have help a model focus on the data points with less uncertainty (such as in meta-analysis).
+
+There are undoubtedly more types of weights in other domains. 
+
+Generally speaking, case weights are usually non-negative. 
+
+## How are they used in base R?
+
+The best example is for categorical data where a small number of possible categories are observed many times. For example, the `UCBAdmissions` contains "Aggregate data on applicants to graduate school at Berkeley for the six largest departments in 1973 classified by admission and sex."
```

Comment:
Remove all references to base R and replace them with Classical Statistical Models. Other statistical systems such as Stata and Minitab have similar constructs: https://support.minitab.com/en-us/minitab/18/help-and-how-to/modeling-statistics/regression/supporting-topics/basics/weighted-regression/#:~:text=Weighted%20regression%20is%20a%20method,a%20constant%20variance%20(homoscedasticity).

Also, rather than "best example" say something like a typical classical example would be for ...